cpp/oneapi/dal/algo/pca/backend/gpu/finalize_train_kernel_cov_dpc.cpp

```diff
@@ -0,0 +1,143 @@
+/*******************************************************************************
+* Copyright 2023 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/algo/pca/backend/gpu/finalize_train_kernel.hpp"
+#include "oneapi/dal/algo/pca/backend/common.hpp"
+#include "oneapi/dal/backend/primitives/lapack.hpp"
+#include "oneapi/dal/backend/primitives/reduction.hpp"
+#include "oneapi/dal/backend/primitives/stat.hpp"
+#include "oneapi/dal/backend/primitives/utils.hpp"
+#include "oneapi/dal/algo/pca/backend/sign_flip.hpp"
+#include "oneapi/dal/table/row_accessor.hpp"
+
+namespace oneapi::dal::pca::backend {
+
+namespace bk = dal::backend;
+namespace pr = oneapi::dal::backend::primitives;
+using alloc = sycl::usm::alloc;
+
+using bk::context_gpu;
+using model_t = model<task::dim_reduction>;
+using task_t = task::dim_reduction;
+using input_t = partial_train_result<task_t>;
+using result_t = train_result<task_t>;
+using descriptor_t = detail::descriptor_base<task::dim_reduction>;
+
+template <typename Float>
+auto compute_eigenvectors_on_host(sycl::queue& q,
```

Comment:
It is possible to compute eigenvectors on device in three steps using oneMKL LAPACK functions:
1. Reduce symmetric correlation matrix to tridiagonal form $A = Q \cdot T \cdot Q^T$ with [?sytrd](https://www.intel.com/content/www/us/en/docs/onemkl/developer-reference-fortran/2023-2/sytrd.html) or ?syrdb
2. Compute eigenvectors $u$ of a tridiagonal matrix T using [?stemr](https://www.intel.com/content/www/us/en/docs/onemkl/developer-reference-fortran/2023-2/stemr.html)
3. Compute eigenvectors $v$ of the original matrix $A$ from the eigenvectors of the matrix $T$ as $v = Q \cdot u$

Eigenvalues of $A$ and $T$ are the same.